templates/llm-router/README.md

```diff
@@ -0,0 +1,1522 @@
+    num_checkpoints_to_keep: 1
+    no_gradient_checkpoint: False
+    output_dir: /mnt/local_storage
+    deepspeed:
+      config_path: config_files/deepspeed/zero_3_optimizer_parameter_offload.json
+    flash_attention_2: true
+    classifier_config:
+      label_tokens:
+          - "[[1]]"
+          - "[[2]]"
+          - "[[3]]"
+          - "[[4]]"
+          - "[[5]]"
+
+
+
+```python
+# View job yaml config
+!cat configs/ft_job.yaml
+```
+
+    name: llm-router-tutorial
+    entrypoint: python src/ft.py configs/ft_config_a10.yaml
+    image_uri: localhost:5555/anyscale/llm-forge:0.5.0.0
+    requirements: requirements.txt
+    max_retries: 0
+
+
+
+```python
+# Job submission
+!anyscale job submit --config-file configs/ft_job.yaml --exclude assets
+```
+
+    [1m[36mOutput[0m[0m
+    [0m[1m[36m(anyscale +1.0s)[0m [0m[0m[0m[0mSubmitting job with config JobConfig(name='llm-router-tutorial', image_uri='localhost:5555/anyscale/llm-forge:0.5.0.0', compute_config=None, env_vars=None, py_modules=None, cloud=None, project=None, ray_version=None).[0m
+    [0m[1m[36m(anyscale +2.5s)[0m [0m[0m[0m[0mUploading local dir '.' to cloud storage.[0m
+    [0m[1m[36m(anyscale +3.5s)[0m [0m[0m[0m[0mJob 'llm-router-tutorial' submitted, ID: 'prodjob_16krca7sgdjyeh2eyf81h6q9uf'.[0m
+    [0m[1m[36m(anyscale +3.5s)[0m [0m[0m[0m[0mView the job in the UI: https://console.anyscale.com/jobs/prodjob_16krca7sgdjyeh2eyf81h6q9uf[0m
+    [0m[1m[36m(anyscale +3.5s)[0m [0m[0m[0m[0mUse `--wait` to wait for the job to run and stream logs.[0m
+    [0m[0m
+
+The job takes around 10 minutes on `4xA100-80gb` and 1 hour on `8xA10-22gb` to finish. Training logs will show the final model checkpoint, e.g.:
+
+```
+Best checkpoint is stored in:
+storage-bucket-cld-tffbxe9ia5phqr1unxhz4f7e1e/org_4snvy99zwbmh4gbtk64jfqggmj/cld_tffbxe9ia5phqr1unxhz4f7e1e/artifact_storage/amjad__almahairi_dkaubsimoyxpiksqxqkxrfgfvzzotwtacs/llmforge-finetuning/meta-llama/Meta-Llama-3-8B/TorchTrainer_2024-06-21_17-02-52/epoch-4
+With perplexity: 1.0318867739521242
+```
+This checkpoint can be used to run batch inference or serve the model online.
+
+# Step 3: Offline Evaluation <a id="offline-eval"></a>
+
+Next, we will conduct an offline evaluation of the model trained on an out-of-domain dataset. The same model, now trained on the full dataset, is available in the following GitHub repository: [https://github.com/lm-sys/RouteLLM/](https://github.com/lm-sys/RouteLLM/), along with other router models.
+
+
+### Install `RouteLLM` package
+
+
+```python
+# Clone the repository under /home/ray/default/
+!git clone https://github.com/lm-sys/RouteLLM.git /home/ray/default/RouteLLM
+
+# Change to the cloned repository directory
+%cd /home/ray/default/RouteLLM
+
+# Install the package with the specified extras
+!pip install -e .[eval]
+```
+
+    fatal: destination path '/home/ray/default/RouteLLM' already exists and is not an empty directory.
```

Comment:
probably don't need to commit the output cells

Reply:
This one skipped me, good catch! Do you suggest I remove all of them? I kept only a summary showing what the user will see, but maybe not important

Reply:
yeah I suggest removing all of them unless there's some really important output to display